src/db.c

```diff
@@ -860,6 +890,47 @@ int parseScanCursorOrReply(client *c, robj *o, unsigned long *cursor) {
     return C_OK;
 }
 
+char *obj_type_name[OBJ_TYPE_MAX] = {
+    "string", 
+    "list", 
+    "set", 
+    "zset", 
+    "hash", 
+    "", /* module type is special */
+    "stream"
+};
+
+long long getObjectTypeByName(char *name) {
+    /* empty string as a unknown type */
+    if (strlen(name) == 0) return LLONG_MAX;
+
+    for (long long i = 0; i < OBJ_TYPE_MAX; i++) {
+        if (!strcasecmp(name, obj_type_name[i])) {
+            return i;
+        }
+    }
+
+    moduleType *mt = moduleTypeLookupModuleByName(name);
```

Comment:
`moduleTypeLookupModuleByName` is using `memcmp`, so moduletype's name is case-sensitive.

But currently `strcasecmp` is used for comparing typename, which may cause a breaking change. In theory, this should be considered a bugfix because different case means a different moduletype, and this situation is also very rare, so it is not a breaking change in practical.

Reply:
good catch!
considering the other type filters are case insensitive, i think we should have this one case insensitive too. for consistency.
besides, it's easier for users to use (having two modules with similar names is probably just a theoretic case).
and also, let's avoid a breaking change if we didn't mean for it and we can avoid it.

bottom line, let's add a boolean argument for moduleTypeLookupModuleByName to be case insensitive.

Reply:
@soloestoy Cool! 👍 
@oranagra Agree. It's better to be consistent for users, and having two modules with similar name is rarer than input case insensitivity when scanning module type. 
But this is still a breaking change for docs , that means we definitely not support case sensitive scan for module type, and the same needs to be stated in the module's documentation. Should we be more careful to make decision?

Reply:
@judeng not sure what you mean.
do we agree that we should avoid the breaking change? and you argue that in addition to that we should document that it ignores case for modules?
i'd rather not document that, and have the opportunity to break it on the next major release so it's consistent with other types.

Reply:
Your understanding is completely correct, I will follow your point.